test/unit/math/mix/fun/eigendecompose_sym_test.cpp

```diff
@@ -0,0 +1,106 @@
+#include <test/unit/math/test_ad.hpp>
+
+TEST(MathMixMatFun, eigendecomposeSym) {
+  auto g = [](const auto& y) {
+    // eigenvectors
+    // maintain symmetry for finite diffs; ignore if not square
+    if (y.rows() != y.cols()) {
+      return std::get<0>(stan::math::eigendecompose_sym(y));
+    }
+    auto a = ((y + y.transpose()) * 0.5).eval();
+    return std::get<0>(stan::math::eigendecompose_sym(a));
+  };
+
+  auto f = [](const auto& y) {
+    // eigenvalues
+    if (y.rows() != y.cols()) {
+      return std::get<1>(stan::math::eigendecompose_sym(y));
+    }
+    auto a = ((y + y.transpose()) * 0.5).eval();
+    return std::get<1>(stan::math::eigendecompose_sym(a));
+  };
```

Comment:
(optional) for these, instead of multiple functions you could just sum each of the results of the function and then return the scalar. But I'm also fine with this

Reply:
I prefer keeping them separate due to the granularity it provides if one derivative is incorrect but the other isn't